dataprofiler/data_readers/filepath_or_buffer.py

```diff
@@ -31,35 +33,43 @@ def __init__(self, filepath_or_buffer, open_method='r', encoding=None, seek_offs
     def __enter__(self):
         if isinstance(self._filepath_or_buffer, str):
             self._filepath_or_buffer = open(
-                self._filepath_or_buffer, self.open_method, encoding=self._encoding)
+                self._filepath_or_buffer, self.open_method,
+                encoding=self._encoding)
 
-        elif isinstance(self._filepath_or_buffer, BytesIO) and self.open_method == 'r':
-            self._filepath_or_buffer = DetachingTextIOWrapper(self._filepath_or_buffer, encoding=self._encoding)
+        elif isinstance(self._filepath_or_buffer, BytesIO) \
+                and self.open_method == 'r':
+            self._filepath_or_buffer = \
+                DetachingTextIOWrapper(self._filepath_or_buffer,
+                                       encoding=self._encoding)
 
         elif not data_utils.is_stream_buffer(self._filepath_or_buffer):
             # Raise AttributeError if attribute value not found.
-            raise AttributeError(f'Type {type(self._filepath_or_buffer)} is invalid. \
-                filepath_or_buffer must be a string or StringIO/BytesIO object')
+            raise AttributeError(f'Type {type(self._filepath_or_buffer)} is '
+                                 f'invalid. filepath_or_buffer must be a '
+                                 f'string or StringIO/BytesIO object')
 
         if self.seek_offset is not None:
             self._filepath_or_buffer.seek(self.seek_offset, self.seek_whence)
 
         return self._filepath_or_buffer
 
     def __exit__(self, exc_type, exc_value, exc_traceback):
-        if isinstance(self._filepath_or_buffer, (StringIO, BytesIO, DetachingTextIOWrapper)):
+        if isinstance(self._filepath_or_buffer, (StringIO, BytesIO,
+                                                 DetachingTextIOWrapper)):
```

Comment:
If we instead on `__init__` stored a variable `self._is_wrapped` could we get rid of `DetachingTextIOWrapper`?

Basically we could check that variable to see if we need to detach.prior to this if. If so, then do a swap so to speak.
```python
wrapper = self._filepath_or_buffer
self._filepath_or_buffer = wrapper.buffer
wrapper.detach()
```